cli/impl/src/main/java/io/helidon/build/cli/impl/ArchetypeInvoker.java

```diff
@@ -279,26 +279,87 @@ Path invoke() throws IOException {
      * Invoker for the archetype V2 engine.
      */
     static class V2Invoker extends ArchetypeInvoker {
+        private static final String ENTRY_POINT_DESCRIPTOR = "flavor.xml";
+        private static final String FLAVOR_PROPERTY = "flavor";
+        private static final String PROJECT_NAME_PROPERTY = "project.name";
+        private static final String GROUP_ID_PROPERTY = "project.groupId";
+        private static final String ARTIFACT_ID_PROPERTY = "project.artifactId";
+        private static final String PACKAGE_NAME_PROPERTY = "package";
+        private static final String HELIDON_VERSION_PROPERTY = "helidon.version";
+        private static final String BUILD_SYSTEM_PROPERTY = "build-system";
+        private static final String ARCHETYPE_BASE_PROPERTY = "base";
+        private static final String SUPPORTED_BUILD_SYSTEM = "maven"; // We only support one
 
         private V2Invoker(Builder builder) {
             super(builder);
         }
 
         @Override
         Path invoke() throws IOException {
-            Path projectDir = projectDirSupplier().apply(initOptions().initProperties().get("name"));
+            InitOptions initOptions = initOptions();
+            Map<String, String> params = new HashMap<>();
+            Map<String, String> defaults = new HashMap<>();
+
+            // We've already got helidon version, don't prompt again
+            params.put(HELIDON_VERSION_PROPERTY, initOptions.helidonVersion());
+
+            // Don't prompt for build system since we only support one for now
+            params.put(BUILD_SYSTEM_PROPERTY, SUPPORTED_BUILD_SYSTEM);
+
+            // Set flavor if provided on command-line
+            if (initOptions.flavorOption() != null) {
+                params.put(FLAVOR_PROPERTY, initOptions.flavorOption().toString());
+            }
+
+            // Set base if provided on command-line
+            if (initOptions.archetypeName() != null) {
```

Comment:
Hello Bryan,

I think this check is always true as archetypeName has a default value in initOption constructor. 
Is it correct ?

Reply:
Yes, you're right, good catch.